js_modules/dagit/src/configeditor/codemirror-yaml/mode.tsx

```diff
@@ -393,21 +394,21 @@ function findAutocomplete(
   parents = parents.filter(({ indent }) => currentIndent > indent);
   const immediateParent = parents[parents.length - 1];
 
-  let available = typeConfig.types[typeConfig.rootTypeName].fields;
+  let available = typeConfig.types[typeConfig.rootTypeKey].fields;
 
   if (available && parents.length > 0) {
     for (const parent of parents) {
       const parentTypeDef = available.find(({ name }) => parent.key === name);
 
-      if (!parentTypeDef || !parentTypeDef.type.name) {
+      if (!parentTypeDef || !parentTypeDef.configType.name) {
         return [];
       }
 
-      let childTypeName = parentTypeDef.type.name;
+      let childTypeName = parentTypeDef.configType.name;
       let childEntiresUnique = true;
 
-      if (parentTypeDef.type.name.startsWith("List.")) {
-        childTypeName = parentTypeDef.type.name.substr(5);
+      if (parentTypeDef.configType.name.startsWith("List.")) {
```

Comment:
we should def do a type check against list here rather than a substring match. List's names will be null soon